Improve logging for skipped event and validate payload before processing

sm43 opened the pull request:
log skipped event as info and log error only for unsupported
controller used to log error for skipped event, now we log skipped
event as info with their id in log, for easy debugging and log
error for unsupported event and when necessary.
&
validate payload for valid json before processing

# Changes

<!-- Describe your changes here- ideally you can get that description straight from
your descriptive commit message(s)! -->

# Submitter Checklist

- [x] ♽  Run `make test lint` before submitting a PR (ie: via [pre-push github hook](../hack/dev/prep-push-hook) no need to waste CPU cycle on CI 
- [ ] 📖 If you are adding a user facing feature or make a change of the behavior, please make sure to document it
- [ ] 🧪 100% coverage is not a target but most of the time we would rather have a unit test if you make a code change.
- [ ] 🎁 If that's something that is possible to do please make sure to check if we can add a e2e test.
- [ ] 🔎 If there is a flakiness in the CI tests then make sure to get the flakyness fixed before merging or if that's not possible there is a good reason to bypass it.

_See [the developer guide](https://github.com/openshift-pipelines/pipelines-as-code/blob/main/docs/development.md) for a bit more details._

Commit:
log skipped event as info and log error only for unsupported

controller used to log error for skipped event, now we log skipped
event as info with their id in log, for easy debugging and log
error for unsupported event and when necessary.

Signed-off-by: Shivam Mukhade <smukhade@redhat.com>

Commit:
validate payload for valid json before processing

Signed-off-by: Shivam Mukhade <smukhade@redhat.com>

sm43 commented:
/retest

sm43 commented:
/retest

chmouel commented:
I wonder if we should just reply something else than 400 : 

![image](https://user-images.githubusercontent.com/98980/161276414-5138fcd6-5a87-4023-a8d2-8c84e34719f7.png)

eventlistenner reply a json ok or something and maybe that would be easier to have this if the user want to check if the controller is up

 (since they probably don't know about the /live thing)

sm43 commented:
> I wonder if we should just reply something else than 400 :
> 
> ![image](https://user-images.githubusercontent.com/98980/161276414-5138fcd6-5a87-4023-a8d2-8c84e34719f7.png)
> 
> eventlistenner reply a json ok or something and maybe that would be easier to have this if the user want to check if the controller is up
> 
> (since they probably don't know about the /live thing)


we could reply
400 (bad request) if it fails payload validation [here](https://github.com/openshift-pipelines/pipelines-as-code/blob/924e735ddf96071f1e92c542fbbd1cd69273d1d0/pkg/adapter/adapter.go#L95)
200 (ok) for ignored request
202 (accepted) for the events which will be processed
wdyt?

chmouel commented:
> 400 (bad request) if it fails payload validation [here](https://github.com/openshift-pipelines>/pipelines-as-code/blob/924e735ddf96071f1e92c542fbbd1cd69273d1d0/pkg/adapter/adapter.go#L95)
> 200 (ok) for ignored request
> 202 (accepted) for the events which will be processed

SGTM! 200 OK with a json output (like eventlistenner does)

chmouel commented:
i still get a 400 
![image](https://user-images.githubusercontent.com/98980/161500753-8703dc52-3d5e-49e7-b745-4152f8b3636d.png)

maybe we can add a e2e test too ? which doesn't need to create a PR or anything, it just test a 200 or 400 when talking directly to the controller..

Commit:
adds e2e test for unsupported/skipped event

Signed-off-by: Shivam Mukhade <smukhade@redhat.com>

Commit:
returns "ok" if not a POST call

the api now returns 200 & "ok" if it is not a POST call
so that users can check if the server is up by a GET call.

Signed-off-by: Shivam Mukhade <smukhade@redhat.com>

chmouel commented on test/invalid_event_test.go:
```diff
@@ -0,0 +1,86 @@
+		t.Fatal("failed to find event listener url")
+	}
+
+	req, err := http.NewRequestWithContext(ctx, "POST", elURL, bytes.NewBuffer(jeez))
```
can you add a test for GET as well please 🙏🏻

sm43 replied:
sry missed it 😅

chmouel commented:
/retest
